Cargo.toml

```diff
@@ -11,13 +11,14 @@ rust-version = "1.60"
 authors = ["Christopher Berner <christopherberner@gmail.com>"]
 
 [lib]
-crate-type = ["cdylib", "rlib"]
+crate-type = ["rlib"]
 
 [dependencies]
 serde = {version = "1.0.102", features=["std", "derive"], optional = true}
 pyo3 = {version = "0.16", features=["extension-module", "abi3-py37"], optional = true }
 wasm-bindgen = {version = "0.2", optional = true}
 js-sys = {version = "0.3.60", optional = true}
+micromath = {version = "2.0.0", optional = true}
```

Comment:
I'd rather not add this dependency. Why is it necessary?

Reply:
This is needed for baremetal, as `core::f32` does not really have `ceil()` and `floor()` functionality. It seems to be best practice solution for no_std purposes. Anyway, this optional crate only includes into `metal` feature builds now.

Reply:
Yeah, I was going to propose this myself. My only concern is that the upstream standard requires floats, which is indeed silly, but would require us to be very careful and flood this point with tests, so I was going to postpone this move a bit, but you are right, it's time. We'll do it soon.

Reply:
Things turned out to be simpler than they seemed; since we do not mess with negative stuff and other fine-grained junk, those operations were replaced by integer division similar to one of [those](https://github.com/rust-lang/rust/issues/88581) that are now waiting to stabilize in `core`.